mapnik/printing.py

```
@@ -785,407 +785,535 @@ def render_legend(self, m, ctx=None, columns=2, width=None, height=None, attribu
-                        legend_text_size = 0
-                        if not item_per_rule:
-                            rule_text = layer_title
-                        if rule_text:
-                            e = self.write_text(
-                                ctx, rule_text, m2pt(
-                                    cwidth - legend_item_box_size[0] - 0.005), 6)
-                            legend_text_size += e[3]
-                            ctx.rel_move_to(0, e[3])
-                        if layer_title in attribution:
-                            e = self.write_text(
-                                ctx,
-                                attribution[layer_title],
-                                m2pt(
-                                    cwidth -
-                                    legend_item_box_size[0] -
-                                    0.005),
-                                6,
-                                fill_color=(
-                                    0.5,
-                                    0.5,
-                                    0.5))
-                            legend_text_size += e[3]
-
-                        if legend_text_size > legend_entry_size:
-                            legend_entry_size = legend_text_size
-
-                        y += legend_entry_size + 2
-                        if y > h:
-                            h = y
-        return (w, h)
+        return default_view
+
+    def add_geospatial_pdf_header(self, m, filename, epsg=None, wkt=None):
+        """
+        Adds geospatial PDF information to the PDF file as per:
+            Adobe® Supplement to the ISO 32000 PDF specification
+            BaseVersion: 1.7
+            ExtensionLevel: 3
+            (June 2008)
+
+        Notes:
+            The epsg code or the wkt text of the projection must be provided.
+            Must be called *after* the page has had .finish() called.
+        """
+        if HAS_PYPDF2 and (epsg or wkt):
```

I think we should raise an error, basically we won't be able to do any of the things we promised with that method. what do you think?

```python
if not HAS_PYPDF2:
    raise RuntimeError("PyPDF2 not available; PyPDF2 required to add geospatial header to PDF")

if not any((epsg,wkt)):
    raise RuntimeError("EPSG or WKT required to add geospatial header to PDF")